src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -5301,6 +5301,222 @@ struct npc_silverpine_generic_actor_fenris_keep : public ScriptedAI
     bool _isWorgen;
 };
 
+enum BloodfangStalker
+{
+    QUEST_EXCISING_THE_TAINT                    = 27181,
+
+    NPC_DARKTUSK_BOAR                           = 46575,
+    NPC_BERARD_THE_MOON_CRAZY                   = 46992,
+
+    SPELL_STALKING                              = 86237,
+    SPELL_SHADOWSTEP                            = 79864,
+    SPELL_KILL_ME_QUEST                         = 86559,
+    SPELL_BLOODY_STRIKE                         = 87359,
+    SPELL_PERMANENT_FEIGN_DEATH_BLOODFANG       = 80636,
+
+    EVENT_SNIFFING                              = 1,
+    EVENT_STRIKE_BOAR                           = 2,
+    EVENT_SHADOWSTEP                            = 4
+};
+
+// 45195 - Bloodfang Stalker
+struct npc_silverpine_bloodfang_stalker : public ScriptedAI
+{
+    npc_silverpine_bloodfang_stalker(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        me->SetPowerType(POWER_ENERGY);
+        me->SetMaxPower(POWER_ENERGY, 100);
+        me->SetPower(POWER_ENERGY, 100, true);
+
+        DoCastSelf(SPELL_STALKING);
+
+        if (me->GetEntry() != NPC_BERARD_THE_MOON_CRAZY)
+            DoCastSelf(SPELL_KILL_ME_QUEST);
+    }
+
+    void Reset() override
+    {
+        _events.Reset();
+
+        DoCastSelf(SPELL_STALKING);
+
+        if (me->GetEntry() != NPC_BERARD_THE_MOON_CRAZY)
+            DoCastSelf(SPELL_KILL_ME_QUEST);
+
+        // Note: I don't like this way of handling. I would rather look for a spawnGUID, but it is not recommended.
+        if (Creature* boar = me->FindNearestCreature(NPC_DARKTUSK_BOAR, 5.0f))
+        {
+            if (boar->HasAura(SPELL_PERMANENT_FEIGN_DEATH_BLOODFANG))
+                _events.ScheduleEvent(EVENT_STRIKE_BOAR, 2s, 12s);
+            else
+                _events.ScheduleEvent(EVENT_SNIFFING, 2s, 12s);
+        }
+        else
+            _events.ScheduleEvent(EVENT_SNIFFING, 2s, 12s);
```

Comment:
If you have sniffs, look for a driver spell that may be a dummy aura targeting anything on which you can just add conditions and a spell script to do this

Reply:
The only aura that they have during which the creature does this is an SPELL_EFFECT_APPLY_AURA with SPELL_AURA_ANIM_REPLACEMENT_SET.